frontend/src/components/MainSearch/MainSearch.js

```diff
@@ -38,6 +40,56 @@ class MainSearch extends Component {
     }
   }
 
+  togglePanel() {
+    this.setState(prevState => ({ open: !prevState.open }));
+  }
+
+  renderChevronButton() {
+    const { open } = this.state;
+
+    if (open) {
+      return (
+        <div className="chevronButtonContainer">
+          <div onClick={this.togglePanel} className="chevron-box">
+            <img
+              src="https://img.icons8.com/ios/26/000000/chevron-down.png"
+              alt="chevron button"
+              className="chevron-image"
+            />
+          </div>
+        </div>
+      );
+    }
+
+    return null;
+  }
+
+  renderSearchInstructions() {
+    const { open } = this.state;
+
+    if (!open) {
+      return (
+        <div className="searchInstrucitons">
+          <p> You can search recipe names or ingredients by keyword. Enter your keywords and click the search button or press enter to search. If you use multiple keywords, the search will return only results containing all of your keywords.</p>
+          <div onClick={this.togglePanel}>
+            <p className="close">Collapse</p>
+          </div>
+        </div>
+      );
+    }
+
+    return null;
+  }
+
+  // renderExplainerPanel() {
+  //   const { open } = this.state;
+  //   if (!open) {
+  //     return <renderChevronButton />;
+  //   }
+
+  //   return <renderSearchInstructions />;
+  // }
```

Comment:
I think if you take the `open` handling out of the individual methods, this approach should be fine. But these functions aren't components, so you'll need to `return renderChevronButton();` instead. I'd also flip these so you're checking `open` instead of `!open`.

Reply:
You could even replace this function with a ternary on line 120, like `{ open ? renderSearchInstructions() : renderChevronButton() }`. Since you've reduced it to one line, a ternary makes sense.